### PR TITLE
UNST 9269 fm1dimp improvements

### PR DESCRIPTION
**Pull request by V-Chavarrias**
# What was done 

<a short description with bullets> 

- UNST-9269: Implemented q-dependent friction coefficient. 
- UNST-9355: Use `realloc`.
- UNST-9359: Implement Manning friction. 
- UNST-9372: Implement Qh boundary condition.
 

# Evidence of the work done 

- [ ]	Video/figures \
<add video/figures if applicable> 
- [x]	Clear from the issue description 
- [ ]	Not applicable 

# Tests 
- [x] Tests updated: `e02_f500_c06_chezy_Q` , `e02_f500_c07_manning`, `e02_f500_c08_qh ` \
<add testcase numbers if applicable, Issue number>
- [ ]	Not applicable 

# Documentation  
- [ ]	Documentation updated \
<add description of changes if applicable, Issue number> 
- [x]	Not applicable 

# Issue link

### CODE REVIEW

File: src/utils_gpl/flow1d_implicit/flow1d_implicit/src/FLBOCH.f90 [hunk 1]
```diff
@@ -442,11 +458,14 @@ subroutine FLBOCH(nbran  ,ngrid  ,branch ,typcr  ,bfrict ,bfricp ,&
...
-                  &0      ,hi     ,qi     ,ui     ,&
-                  &r0     ,c0     )
+                  !call FLCHZT(ibr    ,i      ,nbran  ,ngrid  ,&
+                  !&bfrict ,bfricp ,maxtab ,ntabm  ,&
+                  !&ntab   ,table  ,d90    ,engpar ,&
+                  !&0      ,hi     ,qi     ,ui     ,&
+                  !&r0     ,c0     )
+                
+                  isec_fm=1 ! main section
+                  c0 = getFrictionValue(rgs, spdata, crs%cross(idx_crs)%tabdef, ibr, isec_fm, i, hi, qi, DBLE(ui), DBLE(r0), DBLE(dpt), crs%cross(idx_crs)%chainage)
```
Comment (hrajagers) [inline src/utils_gpl/flow1d_implicit/flow1d_implicit/src/FLBOCH.f90]: Hm ... I was expecting something like `real(ui,dp)` but indeed, `getFrictionValue` still requires `double precision` inputs ... wondering whether D-Flow FM properly converts from `dp` to `double precision` when calling this routine. To discuss with @jjnoort whether the `flow1d` code shouldn't be migrated to `real(dp)` as well (under some other issue than this, but this code will be affected ...).

File: src/utils_gpl/flow1d_implicit/flow1d_implicit/src/FLBOCH.f90 [hunk 2]
```diff
@@ -257,6 +259,10 @@ subroutine FLBOCH(nbran  ,ngrid  ,branch ,typcr  ,bfrict ,bfricp ,&
 !
 !
 !***********************************************************************
+   use m_f1dimp, only: f1dimppar_type    
+   use m_network, only: getFrictionValue
```
[the remark this answers is not in the document]
Comment (V-Chavarrias) [reply src/utils_gpl/flow1d_implicit/flow1d_implicit/src/FLBOCH.f90]: Yes. I think that bit by bit we should restrict SRE to its kernel, and strip away all other things which are done better in FM.

**Comment by V-Chavarrias** [comment]
Thanks, @hrajagers. I made this issue for the conversion to `real(.., dp)`:

https://issuetracker.deltares.nl/browse/UNST-9411
